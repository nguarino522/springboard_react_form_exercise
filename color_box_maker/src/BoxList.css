.BoxList {
    color: white;
}

.BoxList-boxarea {
    display: flex;
    flex-wrap: wrap;
    margin: 10px;
}