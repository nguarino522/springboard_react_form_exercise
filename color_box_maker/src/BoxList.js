import React, { useState } from "react";
import './BoxList.css';
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import { v4 as uuid } from 'uuid';

const BoxList = () => {
    const [boxes, setBoxes] = useState([]);

    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, { ...newBox, id: uuid() }])
    }

    const removeBox = (id) => {
        setBoxes(boxes => boxes.filter(box => box.id !== id))
    }

    return (
        <div className="BoxList">Create a new box by selecting height, width, and color.
            <NewBoxForm addBox={addBox} />
            <div className="BoxList-boxarea">
                {boxes.map( ({id, width, height, backgroundColor}) => {
                    return <Box key={id} id={id} width={width} height={height} backgroundColor={backgroundColor} removeBox={removeBox}/>
                })}                
            </div>
        </div>
    )
}

export default BoxList;