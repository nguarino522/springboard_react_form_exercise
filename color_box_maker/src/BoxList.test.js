import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
    render(<BoxList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

function addBox(boxList, height = "200px", width = "100px", color = "rgb(255, 87, 51)") {
    const heightInput = boxList.getByLabelText("Height (pixels 50-1000):");
    const widthInput = boxList.getByLabelText("Width (pixels 50-1000):");
    const backgroundColorInput = boxList.getByLabelText("Color:");
    fireEvent.change(backgroundColorInput, { target: { value: color } });
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    const button = boxList.getByText("Create Box");
    fireEvent.click(button);
}

it("can add new box", function* () {
    const boxList = render(<BoxList />);
    expect(boxList.queryByText("X")).not.toBeInTheDocument();
    addBox(boxList);
    const removeButton = boxList.getByText("X");
    expect(removeButton).toBeInTheDocument();
    console.log(removeButton.parentElement.style);
    expect(removeButton.parentElement).toHaveStyle("background:", "rgb(255, 87, 51)");
    expect(removeButton.parentElement).toHaveStyle("height:", "200px");
    expect(removeButton.parentElement).toHaveStyle("width:", "200px");
})

it("can remove a box", function () {
    const boxList = render(<BoxList />);
    addBox(boxList);
    const removeButton = boxList.getByText("X");
    expect(removeButton).toBeInTheDocument();
    fireEvent.click(removeButton);
    expect(removeButton).not.toBeInTheDocument();
})