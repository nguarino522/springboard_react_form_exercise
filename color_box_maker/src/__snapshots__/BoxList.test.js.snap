// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    Create a new box by selecting height, width, and color.
    <div
      class="NewBoxForm"
    >
      <form>
        <div
          class="NewBoxForm-form-section"
        >
          <label
            for="height"
          >
            Height (pixels 50-1000): 
          </label>
          <input
            id="height"
            max="1000"
            min="50"
            name="height"
            required=""
            type="number"
            value=""
          />
        </div>
        <div
          class="NewBoxForm-form-section"
        >
          <label
            for="width"
          >
            Width (pixels 50-1000): 
          </label>
          <input
            id="width"
            max="1000"
            min="50"
            name="width"
            required=""
            type="number"
            value=""
          />
        </div>
        <div
          class="NewBoxForm-form-section"
        >
          <label
            for="backgroundColor"
          >
            Color: 
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            type="color"
            value="rgb(0, 0, 0)"
          />
        </div>
        <button>
          Create Box
        </button>
      </form>
    </div>
    <div
      class="BoxList-boxarea"
    />
  </div>
</DocumentFragment>
`;
